server/cluster/coordinator.go

```diff
@@ -933,6 +942,59 @@ func (s *scheduleController) GetDelayUntil() int64 {
 	return 0
 }
 
+const MaxDiagnosisResultNum = 6
+
+type diagnosisManager struct {
+	cluster      *RaftCluster
+	schedulers   map[string]*scheduleController
+	dryRunResult map[string]*cache.FIFO
+}
+
+func newDiagnosisManager(cluster *RaftCluster, schedulerControllers map[string]*scheduleController) *diagnosisManager {
+	return &diagnosisManager{
+		cluster:      cluster,
+		schedulers:   schedulerControllers,
+		dryRunResult: make(map[string]*cache.FIFO),
+	}
+}
+
+func (d *diagnosisManager) diagnosisDryRun(name string) error {
+	if _, ok := d.schedulers[name]; !ok {
+		return errors.Errorf("no scheduler named %s", name)
+	}
+	ops, plans := d.schedulers[name].DiagnoseDryRun()
+	result := newDiagnosisResult(ops, plans)
+	if _, ok := d.dryRunResult[name]; !ok {
+		d.dryRunResult[name] = cache.NewFIFO(MaxDiagnosisResultNum)
+	}
+	queue := d.dryRunResult[name]
+	queue.Put(result.timestamp, result)
+	return nil
+}
+
+type diagnosisResult struct {
+	timestamp       uint64
+	plans           []plan.Plan
+	schedulablePlan []plan.Plan
```

Comment:
What is the difference between `schedulablePlan` and `plans`?

Reply:
I change the name